ampd/src/main.rs

```diff
@@ -44,11 +46,12 @@ enum Output {
 
 #[tokio::main]
 async fn main() -> ExitCode {
+    let metrics_server = start_metrics_server(None).await;
```

Comment:
If you read the warp doc, the `.run` methods says `Run this Server forever on the current thread.`. You are now blocking the entire ampd process to run the server now, which breaks the whole thing completely.